query-algebrizer/src/validate.rs

```diff
@@ -229,4 +274,177 @@ mod tests {
             _ => panic!(),
         };
     }
+
+
+    /// Tests that the top-level form is a valid `not`, returning the clauses.
+    fn valid_not_join(parsed: FindQuery, expected_unify: UnifyVars) -> Vec<WhereNotClause> {
+        // filter all the not clauses
+        let mut nots = parsed.where_clauses.iter().filter(|&x| match *x {
```

Comment:
If you use `into_iter()` here, you won't need to clone below, and you'll have fewer reference operations inside the closure.